tests/integration_test.rs

```diff
@@ -0,0 +1,80 @@
+//! Tests for `get_user` and `post_user` methods
+//! Note: the endpoints for this lib are HARD CODED. only to be used with `https://jsonplaceholder.typicode.com/api/v1/users`
+
+extern crate actix_basic;
+
+use tokio;
+use serde_json::json;
+
+const URL: &str = "https://jsonplaceholder.typicode.com/api/v1/users";
+
+#[tokio::main]
+#[test]
+async fn it_gets_users() -> Result<(),Box<dyn std::error::Error>>{
+
+    let address_check =  actix_basic::Address {
+        street: String::from("Douglas Extension"),
+        suite: String::from("Suite 847"),
+        city: String::from("McKenziehaven"),
+        zipcode: String::from("59590-4157"),
+        geo: actix_basic::Geo {
+            lat: String::from("-68.6102"),
+            lng: String::from("-47.0653"),
+        },
+    };
+
+    let users: Vec<actix_basic::User> = actix_basic::get_users(URL).await?;
+    // println!("---Response: (Address of a user):\n {:#?}", users[2].address.street);
+
+    assert_eq!(address_check.street, users[2].address.street);
+
+    Ok(())
+
+}
+
+#[tokio::main]
+#[test]
+async fn it_posts_a_user() -> Result<(),Box<dyn std::error::Error>>{
+    let user_str = r#"
+    {
+        "name": "Martin Fowler",
+        "email": "martin@martinfowler.com"
+    }"#;
+
+    let resp_json_expected = json!({ "apiId": String::from("v1"), "email": String::from("martin@martinfowler.com"), "id": 11, "name": String::from("Martin Fowler")});
```

Comment:
A more terse way to coerce a &str into a String is to "some value here".into()  (look up the from/into trait in the rust book for more info)

Reply:
idiomatic is `String::from()` based on learning from the Book. Is the standard to use .into() ?

Reply:
.into() is more terse.  It takes the boilerplate from the code so you can more easily scan the code.  The from/into trait is very powerful.